src/Airy.jl

```diff
@@ -0,0 +1,265 @@
+module Airy
+using LinearAlgebra
+using Plots
+
+export airy_k_nicel, airy_nicle_na_intervalu
+
+"""
+    y = airy_premik(y_prej, x_prej, h)
+
+Izračunaj vrednost Airyjeve funkcije za `x_prej`+`h` iz prejšnje vrednosti `y_prej`.
+"""
+function airy_premik(y_prej::Vector, x_prej::Float64, h::Float64=0.01)
+    function A1(xk::Float64, h::Float64)
+        # c = 1/2 - sqrt(3)/6
+        c = 0.2113248654051871177454256097490212721761991
+        tmp = xk + (c * h)
+        return [0 1; tmp 0]
+    end
+
+    function A2(xk::Float64, h::Float64)
+        # c = 1/2 + sqrt(3)/6
+        c = 0.7886751345948128822545743902509787278238008
+        tmp = xk + (c * h)
+        return [0 1; tmp 0]
+    end
+
+    function sigma(h::Float64, A1::Matrix, A2::Matrix)
+        # c = sqrt(3)/12
+        c = 0.1443375672974064411272871951254893639119004
+        return h/2 * (A1 + A2) - c * h*h * kom(A1, A2)
+    end
+
+    function kom(A::Matrix, B::Matrix)
+        return A*B - B*A
+    end
+
+    y = exp(sigma(h, A1(x_prej, h), A2(x_prej, h))) * y_prej
+    return y
+end
+
+"""
+    nicle = airy_k_nicel(k, h)
+
+Izračunaj prvih `k` ničel Airyjeve funkcije, s premiki s korakom `h`.
+"""
+function airy_k_nicel(k::Integer; h::Float64=0.01, metoda::String="bisekcija", max_korakov::Integer=10^12, vrni_vmesne::Bool=false, vrni_povp_iter::Bool=false)
+    nicle = []
+    y_prej1 = 0.35502805388781723926006318600418317639797917419917724058332651030081004245
+    y_prej2 = -0.25881940379280679840518356018920396347909113835493458221000181385610277267
+    y_prej = [y_prej1, y_prej2]
+    step = -h
+    xs_total = [0.0]
+    ys_total = [y_prej[1]]
+    i = 0.0
+    korak = 0
+    total_iter = 0
+    while korak <= max_korakov
+        y = airy_premik(y_prej, i, step)
+        korak = korak + 1
+
+        # najden interval na katerem se nahaja ničla
+        if sign(y[1]) != sign(y_prej[1])
+            # kličemo eno izmed metod, da dobimo ničlo z želeno natančnostjo
+            if metoda == "bisekcija"
```

Comment:
Namesto `if else` bi lahko vse funkcije napisali z isto signaturo in preprosto uporabili argument `metoda`

```rb
n, iter = metoda(y, y_prej, i, step)
```